artists#logical/tic_tac_toe_improved.rb

```diff
@@ -0,0 +1,164 @@
+require 'colorize'
+#Variables generales
+step_jugadas = 0
+array_no_opciones = []
+
+def creation_solution
+    index_final = @num_board**2 - 1
+    @array_jugadas_winner = []
+    
+    arr_vertical = Array.new(@num_board) { Array.new }
+    con_v = 0
+
+    arr_horizontal = []
+
+    arr_diagonal = Array.new(2) { Array.new }
+    
+    for i in 0..index_final
+        #Creando soluciones del michi horizontales
+        arr_horizontal.push(i)
+
+        #Creando soluciones del michi verticales
+        arr_vertical[con_v].push(i)
+        con_v += 1
+
+        #Creando soluciones del michi diagonales
+        arr_diagonal[0].push(i) if i % (@num_board + 1) == 0
+        arr_diagonal[1].push(i) if i > 0 && i < index_final && (i % (@num_board - 1) == 0)
+
+        if (i+1) % @num_board == 0
+            @array_jugadas_winner.push(arr_horizontal)
+            arr_horizontal=[]
+            con_v = 0
+        end
+    end
+    # puts @array_jugadas_winner.to_s
+    # puts arr_vertical.to_s
+    # puts arr_diagonal.to_s
```

Comment:
_Recommendation_: when sending a PR remove messages to the console used to debug your application 😉 
Even if they are commented.

Reply:
Thanks for the data